capture/noworkflow/now/persistence/models/code_component.py

```diff
@@ -136,3 +138,35 @@ def first_evaluation(self):
         """
         from .evaluation import Evaluation
         return self.evaluations.order_by(Evaluation.m.checkpoint).first()
+
+    @classmethod # query
+    def compositions(cls, trial_id):
```

Comment:
The name of this method hints that it returns all compositions, but it is restricted to the ones used in the definition graph that represents the ast. We should think about a new name. Maybe ast_composition?